### PR TITLE
Create documentation for GRE version 1.1.0 and 2.1.0

### PR DESCRIPTION
**Pull request by agrawalkaushik**
- Added section for doing health check for private addresses
- Added section on MSS clamping configuration
- Updated releases

### CODE REVIEW

File: docs/plugin_gre.md [hunk 1]
```diff
@@ -78,7 +78,7 @@ admin@node1.conductor1#
 
 In the above example, you will notice there are two tunnels configured on the router. Each `destination` represents a single tunnel interface and allows the user to configure an IPv4 address for the destination. The `enabled-node` configuration allows the user to control which 128T router node will be operating which tunnel. In addition the `enabled-node > tenant` can be used to apply a specific tenant for the GRE tunnel traffic. For each destination on each enabled-node, the 128T router creates a unique KNI interface and the configured tenant is applied to this interface.
 
-#### Tunnel ICMP health check parameters ####
+### Tunnel ICMP health check parameters ###
```
[the remark this answers is not in the document]
Comment (agrawalkaushik) [reply docs/plugin_gre.md]: 👍 - done

File: docs/plugin_gre.md [hunk 2]
```diff
@@ -96,6 +96,39 @@ Every `link-test-interval` an icmp check is performed to determine the availabil
...
+Some GRE tunnel providers require the endpoint to ping an internal private address to detect the tunnel liveliness. The `icmp-keep-alive > address-type` can be used to configure a private address for keep-alive detection. Consider the following example:
+
+```console
+gre
+    enabled         true
+    plugin-network  10.10.10.0/28
+
+    destination     pri-tunnel
+        name             pri-tunnel
+        host             20.20.20.13
```
Comment (jeffmart) [inline docs/plugin_gre.md:116]: Please use another address in the private range. This address is a real address.

File: docs/plugin_gre.md [hunk 3]
```diff
@@ -347,6 +380,30 @@ config
...
+The TCP MSS (maximum segment size) is the amount of data that the interface is willing to accept in a single TCP segment. This option is negotiated by the client and server based on their local environments. However, tunneling adds extra overhead in terms of packet bytes so its important to be able to adjust the MSS (maximum segment-size) for TCP connections by the routers. By default, the `enforced-mss` is set to be `path-mtu` which allows us to automatically adjust the MSS based on the MTU of the tunnel interface. In addition, user can override the value to a static number as shown in the example below.
+
+```console
+gre
+    enabled         true
+    plugin-network  10.10.10.0/28
+
+    destination     dut3-tunnel
+        name             dut3-tunnel
+        host             20.20.20.13
```
Comment (jeffmart) [inline docs/plugin_gre.md:401]: Same comment as above. If needed point in a note that normally the address would be a public address but the private address is used for this example.

Comment (agrawalkaushik) [reply docs/plugin_gre.md]: done

File: docs/plugin_gre.md (hunk 2 above)
Comment (agrawalkaushik) [reply docs/plugin_gre.md]: done

**Pull request merged**